lib/OrdinaryDiffEqCore/src/integrators/integrator_interface.jl

```diff
@@ -406,6 +407,12 @@ function DiffEqBase.reinit!(integrator::ODEIntegrator, u0 = integrator.sol.prob.
         auto_dt_reset!(integrator)
     end
 
+    if reinit_dae &&
+       (integrator.isdae || SciMLBase.has_initializeprob(integrator.sol.prob.f))
+        DiffEqBase.initialize_dae!(integrator)
+        update_uprev!(integrator)
+    end
```

Comment:
It should probably be after reiniting the cache?

Reply:
I did it here because in `solve` we do it before `initialize_callbacks!`, and running it after seems like it can lead to weird issues